tests/test_backend.py

```diff
@@ -2567,6 +2567,80 @@ def test_selu(self):
         _ = tf.identity(y, name=_TFOUTPUT)
         self._run_test_case([_OUTPUT], {_INPUT: x_val})
 
+    # test for gemm pattern0: alpha*A*B + beta*C
+    def test_gemm_pattern0(self):
+        max_number = 10
+        m = np.random.randint(max_number)
+        n = np.random.randint(max_number)
+        k = np.random.randint(max_number)
+        x_val1 = np.random.rand(m, n).astype("float32")
+        x_val2 = np.random.rand(n, k).astype("float32")
+        x_val3 = np.random.rand(m, k).astype("float32")
+        a = tf.placeholder(tf.float32, x_val1.shape, name=_TFINPUT)
+        b = tf.placeholder(tf.float32, x_val2.shape, name=_TFINPUT1)
+        c = tf.placeholder(tf.float32, x_val3.shape, name=_TFINPUT2)
+        alpha = tf.constant(1.0)
```

Comment:
tf.constant(1.0, dtype=tf.float32); to make sure it's treated as float instead of double.